### PR TITLE
Add BondsPenalty hyperparam

### PR DESCRIPTION
**Pull request by opentaco**
### Add BondsPenalty hyperparam

Given recent issues with (partially) blacklisted validators getting poor dividends, we've been exploring the possibility of adjusting the validation bonds penalty to improve their dividends.

**Bonds penalty**: Weight higher than consensus gets clipped, and bonds calculated from this then also get clipped, which means dividends get clipped. Cabal that sets weights on its own poor servers have their bonds cut and receive less dividends.
**Validation advantage**: Setting weight on the highest performing servers, means that dividends will be relatively better. Cabal that sets weights on its own poor servers will get less dividends because those servers get less incentive.

**Suggestion**: Adjust bonds penalty per netuid, with a default zero penalty, keep validation advantage.
**Expected result**: Partially blacklisted validators that set more weights on a fewer set of servers will enjoy full bonds on those, instead of getting bonds cut like when validation penalty applies.

**Security guarantees**:
A foremost concern is how this will affect the consensus security guarantees, so we simulate that as part of the subtensor integration tests with the `_map_consensus_guarantees()` function, and do a comparison with/without bonds penalty.
We observe as slight erosion of guarantees, at one point quantified as a 73% -> 79% honest utility increase required @ 60% honest stake to retain at least 60% emission. This is acceptable given that we gain significant alleviation of poor dividends due to blacklisting.

### CODE REVIEW

File: pallets/subtensor/src/epoch.rs
Comment (Eugene-hu) [inline pallets/subtensor/src/epoch.rs:167]: So if the bonds penalty = 0.5, will the interpolate function return an average between weights and clipped weights?

**Review by Eugene-hu — APPROVED** [review]
LGTM; when do you think we can release this update taco?

File: pallets/subtensor/src/epoch.rs
Comment (opentaco) [reply pallets/subtensor/src/epoch.rs:167]: Yes, that is correct, the unit tests for `interpolate` has examples of this.

**Comment by opentaco** [comment]
> LGTM; when do you think we can release this update taco?

This is ready for release after outstanding approvals.
@shibshib please merge before the next chain upgrade.

**Comment by opentaco** [comment]
> LGTM; when do you think we can release this update taco?

Once outstanding approvals are granted, it can be merged.
@shibshib please merge before the next chain upgrade.

File: pallets/subtensor/src/lib.rs
Comment (Rubberbandits) [inline pallets/subtensor/src/lib.rs:1974]: Was this benchmarked? If not, will add it to the list of features that need benchmarking before we push latest chain upgrade.

Comment (camfairchild) [reply pallets/subtensor/src/lib.rs:1974]: ~~It was not~~ Edit: not my PR, oops

**Review by Rubberbandits — APPROVED** [review]

File: pallets/subtensor/src/lib.rs
Comment (opentaco) [reply pallets/subtensor/src/lib.rs:1974]: > Was this benchmarked? If not, will add it to the list of features that need benchmarking before we push latest chain upgrade.

No it wasn't benchmarked, but copied the time of another u16 hyperparam.

**Pull request closed without merging**
